src/main/scala/org/broadinstitute/dsde/agora/server/ga4gh/Models.scala

```diff
@@ -1,10 +1,109 @@
 package org.broadinstitute.dsde.agora.server.ga4gh
 
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntity, AgoraEntityType, MethodDefinition}
 import spray.json.DefaultJsonProtocol._
 import spray.json.{DeserializationException, JsString, JsValue, RootJsonFormat}
 
 object Models {
 
+  final val ID_DELIMITER = ":"
+
+  case class ToolId(namespace: String, name: String) {
+    override def toString: String = s"$namespace$ID_DELIMITER$name"
+  }
+  object ToolId {
+    def apply(entity:AgoraEntity) = {
+      assert(entity.namespace.nonEmpty, "cannot create a ToolId if entity namespace is empty")
+      assert(entity.name.nonEmpty, "cannot create a ToolId if entity name is empty")
+      new ToolId(entity.namespace.get, entity.name.get)
+    }
+    def apply(method:MethodDefinition) = {
+      assert(method.namespace.nonEmpty, "cannot create a ToolId if method definition namespace is empty")
+      assert(method.name.nonEmpty, "cannot create a ToolId if method definition name is empty")
+      new ToolId(method.namespace.get, method.name.get)
+    }
+  }
+  case class ToolClass(id: String, name: String, description: String)
+  object ToolClass {
+    def apply(entity:AgoraEntity) = fromEntityType(entity.entityType)
+    def apply(method:MethodDefinition) = fromEntityType(method.entityType)
+    def fromEntityType(entityType: Option[AgoraEntityType.EntityType]) = {
+      assert(entityType.nonEmpty, "cannot create a ToolClass if entity type is empty")
+      val str = entityType.get.toString
+      new ToolClass(str, str, "")
+    }
+  }
+
+  case class Tool(
+    url:String,
+    id: String,
+    organization: String,
+    toolname: String,
+    toolclass: ToolClass,
+    description: String,
+    author: String,
+    `meta-version`: String,
+    contains: List[String],
+    verified: Boolean,
+    `verified-source`: String,
+    signed: Boolean,
+    versions: List[ToolVersion])
+
+  object Tool {
+    def apply(entities:Seq[AgoraEntity]) = {
+      val representative = entities.last
+      new Tool(
+        url="", // TODO
+        id=ToolId(representative).toString,
+        organization="",
+        toolname="", // TODO
+        toolclass=ToolClass(representative),
+        description="", // TODO: should we use a snapshot synopsis or description?
+        author="", // TODO: can we get this from the WDL?
+        `meta-version`="", // TODO
+        contains=List.empty[String],
+        verified=false,
+        `verified-source`="",
+        signed=false,
+        versions=entities.toList map (x => ToolVersion(x))
+      )
+    }
```

Comment:
there's a decent bit of business logic in these `apply` methods. It might make more sense to have e.g.:
```
object Tool {
  def apply(entities:Seq[AgoraEntity]) = {
    toolFromEntities(entities)
  }
}
```
and then break out the `toolFromEntities` into some other class.